File: LeetCode/Java Solutions/Reveal Cards In Increasing Order/reveal.java
import java.util.Deque;
import java.util.LinkedList;
import java.util.Arrays;

class Solution {
    public int[] deckRevealedIncreasing(int[] deck) {
        Arrays.sort(deck);
        Deque<Integer> revealOrder = new LinkedList<>();
        
        for (int i = deck.length - 1; i >= 0; i--) {
            if (!revealOrder.isEmpty()) {
                revealOrder.addFirst(revealOrder.removeLast());
            }
            revealOrder.addFirst(deck[i]);
        }
        
        int[] result = new int[deck.length];
        int idx = 0;
        for (int card : revealOrder) {
            result[idx++] = card;
        }
        return result;
    }
}

File: LeetCode/Python Solutions/Reveal Cards In Increasing Order/reveal.py
from collections import deque

class Solution:
    def deckRevealedIncreasing(self, deck):
        deck.sort(reverse=True)
        revealOrder = deque()
        
        for card in deck:
            if revealOrder:
                revealOrder.appendleft(revealOrder.pop())
            revealOrder.appendleft(card)
            
        return list(revealOrder)
